core/os/time.cpp

```diff
@@ -302,8 +347,7 @@ int64_t Time::get_unix_time_from_datetime_dict(const Dictionary p_datetime) cons
 }
 
 int64_t Time::get_unix_time_from_datetime_string(String p_datetime) const {
-	PARSE_ISO8601_STRING(-1)
-	VALIDATE_YMDHMS(0)
+	PARSE_ISO8601_STRING(0)
```

Comment:
This reverts the change made here https://github.com/godotengine/godot/pull/60215#discussion_r850865198

In master:
* If the string parsing fails due to an incorrect number of elements, it will return -1
* If one of the values in each element is invalid, it will return 0

I think this could be helpful, so I would leave it as-is. But also either way works fine.

> Returned value of `Time::get_unix_time_from_datetime_string`
> - `Time::get_unix_time_from_datetime_string` can't return an error value: both 0 and -1 are valid unix timestamps. Returning a default value instead

The argument is flimsy. It's true that 0 and -1 can both be valid Unix timestamps, but your proposal is to replace both possible return values with just 0 which is also a valid timestamp. That doesn't seem better (but it's also not really worse). In both cases the user can decide to check for an error, or if not, interpret it as a valid value.

If we did only have one return value for invalid input, I think -1 would be better than 0, since it's less likely to be a valid value; if just parsing dates, neither 1970-01-01 or 1969-12-31 would return -1.

@akien-mga Which behavior do you desire? We had it one way, then #60215 changed it, and this PR is proposing to change it back. To be clear this isn't a question of ISO 8601 compliance, this is about what to do when the input is not compliant.

Reply:
> If we did only have one return value for invalid input, I think -1 would be better than 0, since it's less likely to be a valid value

I disagree: just because some return value is less likely to be returned for valid input doesn't automatically signals it as an error, so I prefer to stick to the default value of 0